### ISSUE TITLE
getList() method not available on StoreProductKey

### ISSUE THREAD

**Issue by JohntheFish** [concrete5-community-store/community_store#584]
The core attribute key classes all support a getList() static method.

StoreProductKey does not support that method.

I suspect this is historical and arises from the change in attribute code from c5.7 to v8

Looking at the declarations:
`class StoreProductKey extends Key
`
but in a v8+ core
`class FileKey extends Facade
`
Could it be that by not using the facade, some methods provided by the facade are not available? This is just a guess. The core in this area is too convoluted to be easy to trace just what is happening with a quick dig.

I have not checked, is there also a similar issue for order attributes?

**Comment by Mesuva** [comment]
I suspect you are right in terms of the change from v8. I did have some help from Andrew at one point to get the keys set up correctly, but I reckon things have been updated in the core since then.

Will take a look when I do my next burst of work (which should be pretty soon).

**Comment by Mesuva** [comment]
I've investigated what the difference is here between FileKey and StoreProductKey.

There's actually two FileKey classes defined in concrete5, one is the ORM entity, whilst the other one is a facade class for access to the FileCategory. That one is aliased to FileAttributeKey.

StoreProductKey is an entity class, so should be extending Key, in the same way that Concrete\Core\Entity\Attribute\Key\FileKey does.
So you don't actually access getList() from those classes, you access it from the facade class.

What we haven't had in Community Store are those facade class to make accessing such functionality a bit cleaner.
Their ommission doesn't stop you from doing the same things, it's just a bit more longwinded, for example to get all the product attributes you go:
```
 $productCategory = $this->app->make('Concrete\Package\CommunityStore\Attribute\Category\ProductCategory');
 $attrList = $productCategory->getList();
```
But I like having things done the same way as the core, so I've just pushed up to the master branch some facade classes, replacing the need to do the above.
https://github.com/concrete5-community-store/community_store/commit/e5e5f82ae0c065dbaf114e3be5c852e874ad77cb
